Mathlib/Data/Vector/Basic.lean

```diff
@@ -736,4 +773,39 @@ instance : IsLawfulTraversable.{u} (flip Vector n) where
 --       q(α) q(n) q(x) ih
 -- #align vector.reflect vector.reflect
 
+@[simp]
+theorem replicate_succ : replicate (n+1) val = val ::ᵥ (replicate n val) :=
+  rfl
+
+@[simp]
+theorem get_append_cons_zero : get (append (x ::ᵥ xs) ys) ⟨0, by simp⟩ = x :=
```

Comment:
```suggestion
theorem get_append_cons_zero : (append (x ::ᵥ xs) ys)[0] = x :=
```
We probably want to standardize on a simp-normal form here, either `get` or `·[·]`.  For arrays we simplify everything to `·[·]` so I think this would be good here as well.  [@[simp] theorem get_eq_getElem (a : Array α) (i : Fin _) : a.get i = a[i.1] := rfl](https://github.com/leanprover/std4/blob/d5471b83378e8ace4845f9a029af92f8b0cf10cb/Std/Data/Array/Lemmas.lean#L40-L41)

Reply:
> We probably want to standardize on a simp-normal form here, either `get` or `·[·]`. For arrays we simplify everything to `·[·]` so I think this would be good here as well. [@[simp] theorem get_eq_getElem (a : Array α) (i : Fin _) : a.get i = a[i.1] := rfl](https://github.com/leanprover/std4/blob/d5471b83378e8ace4845f9a029af92f8b0cf10cb/Std/Data/Array/Lemmas.lean#L40-L41)

That does seem more consistent; do we then want to change all the existing `get_...` lemmas for `Vector` in terms of `·[·]`?

Reply:
My personal preference is for `·[·]`, and I think that `Vector` is closer to `Array` than to `List`.  We generally don't access lists by index.

I don't remember any discussion about the simp-normal form for indexing.  Maybe it's best to take this to Zulip.

Reply:
I started a topic on Zulip: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simp.20normal.20form.20for.20.60Vector.60.20indexing.3A.20.60get.60.20vs.20.60.2E.5B.2E.5D.60

Reply:
Still, do you think we should change existing theorems in the current PR, or make the changes in a subsequent PR, so that it's a bit more clear which changes are new API, and which changes are just related to the new simp normal form?